examples/src/bin/rpk_example.rs

```diff
@@ -0,0 +1,390 @@
+use log::trace;
+use pki_types::{CertificateDer, PrivateKeyDer, ServerName, SubjectPublicKeyInfoDer, UnixTime};
+use rustls::client::danger::{HandshakeSignatureValid, ServerCertVerified, ServerCertVerifier};
+use rustls::crypto::WebPkiSupportedAlgorithms;
+use rustls::crypto::{
+    aws_lc_rs as provider, verify_tls12_signature_with_spki, verify_tls13_signature_with_spki,
+};
+use rustls::server::danger::{ClientCertVerified, ClientCertVerifier};
+use rustls::version::TLS12;
+use rustls::{
+    ClientConfig, ClientConnection, DigitallySignedStruct, DistinguishedName, ServerConfig,
+    ServerConnection, SignatureScheme, Stream,
+};
+use std::time::Duration;
+use std::{
+    env,
+    fs::File,
+    io::{BufReader, Read, Write},
+    net::{TcpListener, TcpStream},
+    result::Result,
+    sync::Arc,
+};
+use std::{io, thread};
```

Comment:
I think these imports need to be updated to match our style guide, both in terms of ordering and also in terms of how items are grouped with parent modules.

Reply:
There is a tendency in some Rust projects to gather a boatload of relatively niche examples which will then incur maintenance overhead on things like dependency upgrades, so I've started pushing back on adding more examples for relatively niche things in those projects. Feels like this might be on of those things, and this could live in some other repo for people who care about raw keys?

Reply:
I haven't felt like our `examples/` dir has been a source of much maintenance burden (it feels relatively rare that I touch files in there except for breaking API changes). That said, I agree raw public keys is likely to be quite niche and so wouldn't be heartbroken if the example were kept external to the project.

@ctz Do you have a pref on this subject?